src/components/AddBoardCard.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from "react";
+import Modal from "react-modal";
+import { useDispatch } from "react-redux";
+import { addBoard } from "../store/slices/boardSlice";
+import * as styles from "../styles/addBoardCard.css";
+import { logActivity } from "../store/slices/activityLogSlice";
+
+Modal.setAppElement("#root");
+
+const AddBoardCard: React.FC = () => {
+  const dispatch = useDispatch();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [boardTitle, setBoardTitle] = useState("");
+
+  const handleOpenModal = () => {
+    setIsModalOpen(true);
+  };
+
+  const handleCloseModal = () => {
+    setIsModalOpen(false);
+    setBoardTitle(""); // Clear input on close
```

Comment:
To avoid doing this manually check if your library allows unmounting on close. Saves on memory too.